content/5_Plat/Geo_Pri.mdx

```diff
@@ -403,6 +403,72 @@ int main() {
 
 - [My Templates](<https://github.com/bqi343/USACO/tree/master/Implementations/content/geometry%20(13)/Primitives>)
 
+## Angles
+
+### Maximum Number of Visible Points
+
+<FocusProblem problem="radians" />
+
+Working with angles only makes sense in the context of lines and segments. Therefore, let's consider the segments
+connecting the initial position and each dot: $(pos_x, pos_y)-(x, y)$. One such segment is inside the field of
+view if and only if it's slope is bigger than the lower bound and smaller than the upper bound. Since we're
+working with angles, we'll convert the slopes into radians.
+
+Consider $\alpha$ as the angle formed by one line with the $\texttt{ox}$ axis. Then we have:
+$$
+\texttt{slope}=\frac{\Delta y}{\Delta x}=\tg{\alpha} \\
+\alpha=\arctg{\frac{\Delta y}{\Delta x}}
+$$
+
+The movement of the field of view can be seen as a [sliding window](/gold/sliding-window) applied
+on the sorted angles of the points. The angle of the field of vies could be large enough to see
+points from the beginning of the sorted array, thus the array of angles should be duplicated - cyclic array.
+
+#### Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+class Solution {
+  public:
+	int visiblePoints(vector<vector<int>> &points, int angle,
+	                  vector<int> &location) {
+		int cnt = 0;
```

Comment:
rename

Reply:
What do you have againt `cnt` ? You always ask me to change it :))

Reply:
Renamed

Reply:
don't see it on the pr? or did you just forget to push lol

Reply:
lol i always comment before pushing so i don't forget

Reply:
maybe i should stop doing it